fix skipping of cpp attributes.

tsteven4 opened the pull request:
when skipping a CPP attribute skipCPPOrAlignAttribute returns a token for the last ']'  in '[[xxxx]]'.  This works out well with iterators in loops, but wasn't accounted for when scanning ahead when multiple CPP attributes exist, e.g.
 `[[noreturn]] [[gnu::format(printf, 1, 2)]] void fatal(const char*, ...);`
As a result the noreturn attribute above won't be set, e.g. in 2.10.3.

This seems worthy of a test case, but I haven't figured out how your tests work.

danmar commented:
Can you look in `TestTokenizer::simplifyCPPAttribute` if it would make sense to add some more test case there?

tsteven4 commented:
> Can you look in `TestTokenizer::simplifyCPPAttribute` if it would make sense to add some more test case there?

@danmar good call, I added some tests to make sure the simplification works.  I already added a test to verify the noreturn attribute gets applied to the function.

danmar commented:
@tsteven4 Thanks for your contribution! I wonder what name would you like that we add in our AUTHORS?
